VigilanceMeteoFrance/VigilanceMeteoFrance/Program.cs

```diff
@@ -0,0 +1,157 @@
+﻿using Constellation;
+using Constellation.Package;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Xml;
+using System.Xml.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace VigilanceMeteoFrance
+{
+    public class Program : PackageBase
+    {
+        /// <summary>
+        /// Refresh rate of the vigilance.
+        /// </summary>
+        public int RefreshInterval { get; set; }
+
+        /// <summary>
+        /// Departements' number to watch.
+        /// </summary>
+        public string Departement { get; set; }
+
+        static void Main(string[] args)
+        {
+            PackageHost.Start<Program>(args);
+        }
+
+        public override void OnStart()
+        {
+            PackageHost.WriteInfo("Package starting - IsRunning: {0} - IsConnected: {1}", PackageHost.IsRunning, PackageHost.IsConnected);
+            this.RefreshInterval = PackageHost.GetSettingValue<int>("RefreshInterval");
+            this.Departement = PackageHost.GetSettingValue<string>("Departement");
+            if (Departement!="00")
+            {
+                Task.Factory.StartNew(() =>
+                {
+                    while (PackageHost.IsRunning)
+                    {
+                        var elements = Departement.Split(new[] { ',' }, System.StringSplitOptions.RemoveEmptyEntries);
+                        foreach (string items in elements)
+                        {
+                            Vigilance Test = GetVigilance(items);
+                            PackageHost.PushStateObject<Vigilance>(items, Test);
+                        }
+                        int wait = RefreshInterval * 60000;
+                        Thread.Sleep(wait);
+                    }
+                }, TaskCreationOptions.LongRunning);
+            }
+        }
+
+        /// <summary>
+        /// Get vigilance for a departement.
+        /// </summary>
+        [MessageCallback]
+        public Vigilance GetVigilance(string Departement)
+        {
+            string url = @"http://vigilance.meteofrance.com/data/NXFR33_LFPW_.xml";
```

Comment:
Put this string in a configuration variable, to avoid recompiling when an url change occured.